docs/dev/dev_dagster.rst

```diff
@@ -0,0 +1,96 @@
+.. _dev_dagster:
+
+===============================================================================
+Developing with Dagster
+===============================================================================
+
+Reloading code locations
```

Comment:
How is reloading a code location different from "Reloading defs" (which is sometimes a button in the upper-right hand corner of the page).

Does the code location only need to be updated when a NEW asset is added (or removed?) Or also when an existing asset definition is changed (by which I mean the content of the `@asset` decorator is changed, or an asset factory method's contents are changed)?

What kinds of changes to the code can we do and still have fastest/easiest option of rematerializing an existing asset give us the updated output?

Reply:
Good question. I think only need to reload code location when you change the structure of the DAG (add, remove or change asset dependencies). I created a [dagster discussion](https://github.com/dagster-io/dagster/discussions/13819) to get some clarity.